File: src/test/java/kitchenpos/domain/FakeOrderTableRepository.java
package kitchenpos.domain;

import kitchenpos.order.domain.OrderTable;
import kitchenpos.order.domain.OrderTableRepository;

import java.util.*;

public class FakeOrderTableRepository implements OrderTableRepository {

    private Map<UUID, OrderTable> inMemory = new HashMap<>();

    @Override
    public OrderTable save(OrderTable orderTable) {
        inMemory.put(orderTable.getId(), orderTable);
        return orderTable;
    }

    @Override
    public Optional<OrderTable> findById(UUID orderTableId) {
        return Optional.ofNullable(inMemory.get(orderTableId));
    }

    @Override
    public List<OrderTable> findAll() {
        return inMemory.values().stream().toList();
    }
}

File: src/test/java/kitchenpos/eatinorders/tobe/FakeOrderTableRepository.kt
package kitchenpos.eatinorders.tobe

import kitchenpos.eatinorders.tobe.domain.OrderTable
import kitchenpos.eatinorders.tobe.domain.OrderTableRepository
import java.util.*

object FakeOrderTableRepository : OrderTableRepository {
    private val orderTableMap: MutableMap<UUID, OrderTable> = mutableMapOf()

    override fun save(orderTable: OrderTable): OrderTable {
        orderTableMap[orderTable.id] = orderTable

        return orderTable
    }

    override fun findById(orderTableId: UUID): Optional<OrderTable> {
        return Optional.ofNullable(orderTableMap[orderTableId])
    }
}
